presto-orc/src/main/java/com/facebook/presto/orc/reader/FloatStreamReader.java

```diff
@@ -174,12 +175,180 @@ public String toString()
+                if (nextActive == truncationRow) {
+                    break;
+                }
+                if (presentStream != null && !present[i]) {
+                    if (filter == null || filter.testNull()) {
+                        if (filter != null) {
+                            output.append(i + posInRowGroup, activeIdx);
+                        }
+                        addNullResult();
+                    }
+                }
+                else {
+                    // Non-null row in qualifying set.
+                    if (toSkip > 0) {
+                        dataStream.skip(toSkip);
+                    }
+                    float value = dataStream.next();
+                    if (filter != null) {
+                        if (filter.testFloat(value)) {
+                            output.append(i + posInRowGroup, activeIdx);
+                            addResult(floatToRawIntBits(value));
+                        }
+                    }
+                    else {
+                        // No filter.
+                        addResult(floatToRawIntBits(value));
+                    }
+                }
+                if (++activeIdx == input.getPositionCount()) {
+                    toSkip = countPresent(i + 1, end - posInRowGroup);
+                    break;
+                }
+                nextActive = inputPositions[activeIdx];
+                if (outputChannel != -1 && numResults > resultSizeBudget) {
+                    truncationRow = inputQualifyingSet.truncateAndReturnTruncationRow(activeIdx);
+                }
+                continue;
+            }
+            else {
+                // The row is not in the input qualifying set. Add to skip if non-null.
+                if (presentStream == null || present[i]) {
+                    toSkip++;
+                }
+            }
+        }
+        if (toSkip > 0) {
+            dataStream.skip(toSkip);
+        }
+        endScan(presentStream);
+    }
+
+    private void addResult(int value)
+    {
+        if (outputChannel == -1) {
+            return;
+        }
+
+        int position = numValues + numResults;
+        ensureValuesCapacity(position + 1, false);
+        values[position] = value;
+        if (valueIsNull != null) {
+            valueIsNull[position] = false;
+        }
+        numResults++;
+    }
+
+    @Override
+    protected void ensureValuesCapacity(int capacity, boolean includeNulls)
+    {
+        if (values.length < capacity) {
+            values = Arrays.copyOf(values, Math.max(capacity + 10, values.length * 2));
+            if (valueIsNull != null) {
+                valueIsNull = Arrays.copyOf(valueIsNull, values.length);
+            }
+        }
+        if (includeNulls && valueIsNull == null) {
+            valueIsNull = new boolean[values.length];
+        }
+    }
+
+    @Override
+    public int getFixedWidth()
+    {
+        return REAL.getFixedSize();
```

Comment:
It seems like these could be encapsulated in the super type

Reply:
Some classes have more complicated logic, e.g. SliceDictionaryStreamReader.

Reply:
Can we override for the complicated ones?